libcudacxx/include/cuda/std/__utility/unreachable.h

```diff
@@ -22,32 +22,35 @@
 
 _LIBCUDACXX_BEGIN_NAMESPACE_STD
 
-_CCCL_NORETURN _LIBCUDACXX_HIDE_FROM_ABI void unreachable()
+_CCCL_NORETURN _CCCL_HIDE_FROM_ABI _CCCL_HOST void __unreachable_host_impl()
```

Comment:
I believe we can avoid a separate function call by just turning this into the macro we had before

Reply:
I've got rid of the `NV_IF_ELSE_TARGET` and used the slightly modified version of the original macro.